packages/client/src/authentication/adapters/Credentials.ts

```diff
@@ -0,0 +1,52 @@
+import { DateUtils, invariant } from '@lens-protocol/shared-kernel';
+import jwtDecode, { JwtPayload } from 'jwt-decode';
+
+import { ClockSkewedError } from '../../consts/errors';
```

Comment:
nit: I don't think grouping things by their types scales much. Better to colocate things that change at the same time and for the same reason. The error should live in this module IMO.